### ISSUE TITLE
Freezer rack positions

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#1586]
Why don't freezer racks have positions? Currently, in order to "create" positions in a freezer rack we are barcoding the slots in the rack. Shouldn't we just assign positions to freezer racks the same we do in freezers for the racks?

![capture](https://user-images.githubusercontent.com/5725767/41989374-525ef3be-79fc-11e8-9878-dd3039bbacd6.JPG)

We will need different sizes of racks as some hold 12 freezer boxes and others hold 6 and perhaps others hold other quantities. I understand that there may be odd freezers where the standard 33 positions doesn't hold true, but for those that do, I don't think we should be barcoding positions.

**Comment by dustymc** [comment]
>Why don't freezer racks have positions?

They can - so can anything else.

The position form is somewhat limited - it's a specialized form, intended to deal with specific types of position-holding containers. It can probably be expanded to deal with freezer racks, but I'm not sure how that should be prioritized - they're highly variable, don't (shouldn't!) get created very often, etc. I think most people just bulkload them - the bulkloader has some special handlers for just that.

Not sure what's going on with your second screenshot - looks like a container with positions to me.

**Comment by Jegelewicz** [comment]
The positions are actually barcodes that are scanned into the container (which doesn't have positions). Again, if I select "positions" for this container I get the box of red text above.

The problem with this method is that you can't just move all the boxes from one rack to another, you have to scan them individually into the barcoded "positions".

**Comment by Jegelewicz** [comment]
Shouldn't it work this way?

Freezer - has a barcode and a number of positions (freezers are also highly variable and different kinds of racks will mean different position layouts) and is placed in a room.

Rack - has a barcode and a number of positions and is placed into a position in a freezer.

Freezer box - has a barcode and a number of positions and is placed in a position in a freezer rack.

Cryovial - has a barcode and contains a part or parts and is placed in a position in a freezer box.

In this system, barcodes are placed on physical objects - positions are spaces in those objects. The way we are working right now, sometimes spaces are positions and sometimes they are barcodes. I think it is what causes the difficulty in trying to move everything from one freezer into another while keeping positions constant.

**Comment by dustymc** [comment]
>The positions are actually barcodes that are scanned into the container (which doesn't have positions).

I don't know what this means. Position is a container_type. Like all containers, they can have a barcode.

>if I select "positions" for this container I get the box of red text above.

That is the specialized form which doesn't know what to do with this arrangement. It's only one way of working with a very specific arrangement. It's build for things that don't have barcodes - like wells in a freezer box - although it's been extended past that.

>you can't just move all the boxes from one rack to another

You can bulkload that - the same path you'd take if there were no positions.

>Freezer - has a barcode

It can

> and a number of positions

Some do. (Others hold sacks of mammoth fur and such.)

>Rack - has a barcode

Some do.

>and a number of positions 

Some do.

>is placed into a position in a freezer.

Some are. (Others are on the "spare rack" shelf etc.)

>Freezer box - has a barcode

some do

>and a number of positions 

some do - some hold mummified birds and such

>is placed in a position in a freezer rack.

or on the "spare box" shelf or directly in a rack or wherever else it happens to be....

>Cryovial - has a barcode

some do

>and contains a part or parts 

usually...

> is placed in a position in a freezer box.

sometimes.

>barcodes are placed on physical objects

That's not a limitation, but it probably is/should be true. (Sorta - freezer position barcodes are printed on "maps" stuck to the freezer sometimes.)

>sometimes spaces are positions and sometimes they are barcodes.

that's 100% procedural - you can make most anything most anything. Positions SHOULD be things that can't move. http://arctos.database.museum/info/ctDocumentation.cfm?table=CTCONTAINER_TYPE

>difficulty in trying to move everything from one freezer into another while keeping positions constant.

Positions were designed to facilitate that (and under the assumption that the racks in the recently-dead freezer might be spread out to a bunch of other freezers - eg, a couple end up tossed on top of the sack of mammoth fur and scanned directly into the freezer)

**Comment by Jegelewicz** [comment]
Would it be possible to create a position form for a freezer rack that has 12 positions?

Dimensions of the rack are:
Height: 670mm
Width: 140mm
Depth: 143mm

**Comment by campmlc** [comment]
This would solve our immediate problem for 1,500 existing racks:

Please add:
 "freezer rack, 12 positions- 12 columns"
"freezer rack, 11 positions, 11 columns"
"freezer rack, 8 positions, 8 columns"
"freezer rack, 6 positions, 6 columns"

 to the following list:


This application won't do what you want to do.

You must have a container with positons. (Edit Container, set number of
positions.)
What this form will scan into or create positions for:

   - freezer box - 100 positions - 10 rows, 10 columns, holds *cryovial*
   - freezer box - 81 positions - 9 rows, 9 columns, holds *cryovial*
   - freezer box - 25 positions - 5 rows, 5 columns, holds *cryovial*
   - freezer box - 169 positions - 13 rows, 13 columns, holds *cryovial*
   - freezer - 48 positions - 12 rows, 4 columns, holds *freezer rack*
   - freezer - 33 positions - 11 rows, 3 columns, holds *freezer rack*
   - slide box - 100 positions - 50 rows, 2 columns, holds *slide*

If you have that and you're still getting this error, contact us
<https://arctos.database.museum/contact.cfm>.

**Comment by campmlc** [comment]
"holds* freezer box*" for all.

**Comment by dustymc** [comment]
> immediate problem for 1,500 existing racks:

Why not bulkload those??

**Comment by campmlc** [comment]
Because we need a visual form that everyone can use, including students,
and that can allow multiple boxes to be scanned into a form at once instead
of one at a time into each position. The latter creates the risk that a
student will misread the position.

**Comment by campmlc** [comment]
While we are at it, we need to change the existing positions form for a
freezer with 33 racks so that it is transposed to reflect the actual
positions in a chest freezer:

To:

1 Barcode: 4 Barcode: 7 Barcode: 10 Barcode: 13 Barcode: 16 Barcode:
19 Barcode: 22 Barcode: 25 Barcode: 28 Barcode: 31 Barcode:
2 Barcode: 5 Barcode: 8 Barcode: 11 Barcode: 14 Barcode: 17 Barcode:
20 Barcode: 23 Barcode: 26 Barcode: 29 Barcode: 32 Barcode:
3 Barcode: 6 Barcode: 9 Barcode: 12 Barcode: 15 Barcode: 18 Barcode:
21 Barcode: 24 Barcode: 27 Barcode: 30 Barcode: 33 Barcode:



From

1 Barcode:
2 Barcode:
3 Barcode:
4 Barcode:
5 Barcode:
6 Barcode:
7 Barcode:
8 Barcode:
9 Barcode:
10 Barcode:
11 Barcode:
12 Barcode:
13 Barcode:
14 Barcode:
15 Barcode:
16 Barcode:
17 Barcode:
18 Barcode:
19 Barcode:
20 Barcode:
21 Barcode:
22 Barcode:
23 Barcode:
24 Barcode:
25 Barcode:
26 Barcode:
27 Barcode:
28 Barcode:
29 Barcode:
30 Barcode:
31 Barcode:
32 Barcode:
33 Barcode:

**Comment by dustymc** [comment]
I may have finally figured out what you're talking about. You want (have??) racks WITHOUT barcoded positions, correct?

>change the existing

New Issue - that sort of thing needs to be discussed by itself

**Comment by campmlc** [comment]
Yes, please. We need forms to create box positions just like we have forms to create the rack positions and cryovial positions. And we need to adjust the existing form for the 33 rack freezer rack positions to match the actual positions in a freezer so this tool can be used.

Related issues

See #1588 .

Also, looks like my copy paste didn't work earlier. Retrying (again, what we actually see is very important!)

[Freezer rack positions for 33 racks.xlsx](https://github.com/ArctosDB/arctos/files/2143020/Freezer.rack.positions.for.33.racks.xlsx)

**Comment by campmlc** [comment]
We already have other positions without barcodes, eg freezer rack and cryovial positions. This is just a request to add another position form. Should not require separate issue.

**Comment by Jegelewicz** [comment]
When we create a freezer and add positions, we get this:
![freezer positions](https://user-images.githubusercontent.com/5725767/42002941-5c986e80-7a26-11e8-8675-b273a4fe50a4.JPG)
And we can scan our freezer rack barcodes into the positions.

When we create a freezer rack and add postions we get the big red text box in my initial post. Why? My freezer rack looks like this:
![img_4193](https://user-images.githubusercontent.com/5725767/42003025-aeed8544-7a26-11e8-8c99-2ce3bc494fd4.JPG)
So why don't I get something like this when I add positions?
![freezer rack positions](https://user-images.githubusercontent.com/5725767/42003174-530d1a5e-7a27-11e8-92b7-ddd1d7bf6d7e.jpg)

**Comment by Jegelewicz** [comment]
AND BTW, the freezer form is not functional because a freezer looks like this to a student:
![img_4192](https://user-images.githubusercontent.com/5725767/42003256-c2af5eee-7a27-11e8-8ad9-866da6c70a3d.JPG)
the form should look just like the container.
![freezer positions new](https://user-images.githubusercontent.com/5725767/42003550-378c5b4e-7a29-11e8-8d94-b2c73f4bc0b7.jpg)
Apologies for my rushed Paint job...

**Comment by Jegelewicz** [comment]
This we have resorted to wasting barcodes on both freezer and rack positions which also causes problems for students moving freezer boxes from one place to another.

**Comment by dustymc** [comment]
>Should not require separate issue.

I think you're asking to replace an existing (arbitrary) grid system which is used to track stuff in other collections with a new (still arbitrary) grid system which will break all existing data. If that's the case, it needs a new Issue. Could be I'm still just not understanding something.

I'm not sure what the Excel file is trying to do.

I need linguistic tools to sort linguistically. https://github.com/ArctosDB/arctos/issues/104

>big red text box in my initial post. Why?

Two reasons: 

1. The form isn't built to handle whatever you're trying to do
2. The data weren't created by that form, which requires very specific values in a bunch of places to function.

>wasting barcodes

I would expect scanning pairs of barcodes to result in better data, but I can't back that up.

**Comment by campmlc** [comment]
Maybe we need to schedule a discussion on this topic to clear up some of
the misunderstanding. The tools already exist for this - nothing new is
being requested, just an extension.

On Jun 27, 2018 5:50 PM, "dustymc" <notifications@github.com> wrote:

Should not require separate issue.

I think you're asking to replace an existing (arbitrary) grid system which
is used to track stuff in other collections with a new (still arbitrary)
grid system which will break all existing data. If that's the case, it
needs a new Issue. Could be I'm still just not understanding something.

I'm not sure what the Excel file is trying to do.

I need linguistic tools to sort linguistically. #104
<https://github.com/ArctosDB/arctos/issues/104>

big red text box in my initial post. Why?

Two reasons:

   1. The form isn't built to handle whatever you're trying to do
   2. The data weren't created by that form, which requires very specific
   values in a bunch of places to function.

wasting barcodes

I would expect scanning pairs of barcodes to result in better data, but I
can't back that up.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/ArctosDB/arctos/issues/1586#issuecomment-400865789>,
or mute
the thread

**Comment by dustymc** [comment]
Yes, we can do voice. A clear explanation of what you're trying to do would be very useful.

**Comment by Jegelewicz** [comment]
Reviving because this is becoming a roadblock to a grant-funded project.

I'd like to step back and keep this very simple for the moment. Let's start with the freezer positions as seen [here](http://arctos-test.tacc.utexas.edu/containerPositions.cfm?container_id=18085477).

This form does not reflect the actual positions of racks in our freezers. It may do so for other collections since no one else is complaining (@KyndallH ?), but we would like a 33 position freezer form that reflects the actual positions in our freezers. Which look like this:
![image](https://user-images.githubusercontent.com/5725767/46684565-6bea4d00-cbb0-11e8-898e-387cf3474c7b.png)
So, the form should look like this:
![image](https://user-images.githubusercontent.com/5725767/46684626-a05e0900-cbb0-11e8-807c-c376fb5865bb.png)

Matching the "map" of positions in Arctos to the physical positions in the freezer makes it easier for people (students in particular) to find a rack.

We would like this resolved by October 25, 2018

**Comment by dustymc** [comment]
To reiterate:

>I think you're asking to replace an existing (arbitrary) grid system which is used to track stuff in other collections with a new (still arbitrary) grid system which will break all existing data. If that's the case, it needs a new Issue. Could be I'm still just not understanding something.

**Comment by campmlc** [comment]
How is this any different from other positions forms? Are collections not
allowed to have containers with different position arrangements? If so,
that is a fundamental flaw in the model.

MSB uses a different grid for freezer racks than UAM. No institution should
be forced to convert to another institutions system just because we are
incapable if providing flexibility in the forms.

On Oct 9, 2018 12:11 PM, "dustymc" <notifications@github.com> wrote:

To reiterate:

I think you're asking to replace an existing (arbitrary) grid system which
is used to track stuff in other collections with a new (still arbitrary)
grid system which will break all existing data. If that's the case, it
needs a new Issue. Could be I'm still just not understanding something.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/ArctosDB/arctos/issues/1586#issuecomment-428293851>,
or mute
the thread

**Comment by Jegelewicz** [comment]
OK - I have moved the freezer form to its own issue. Now let us return to the original request in this issue.

We would like to have a position form for a freezer rack.

Dimensions of the rack:
Height: 670mm
Width: 140mm
Depth: 143mm

Number of positions = 12

Form should appear as follows:
![freezer rack positions](https://user-images.githubusercontent.com/5725767/42003174-530d1a5e-7a27-11e8-92b7-ddd1d7bf6d7e.jpg)

**Comment by dustymc** [comment]
This is for racks which DO NOT have position barcodes and hold freezer boxes - right?

Maybe we should schedule a quick chat - some comments up ^^ there make me think you do have barcodes (in which case this can't work) and some others once made me briefly think I knew what's going on here...

**Comment by campmlc** [comment]
These racks hold freezer boxes and do not have positions. Because we did
not have this form, we have attempted to physically put barcodes on some
racks for positions. But this is difficult to do on many racks, which don't
have a space for a label. So for some racks we had to give up and not use
positions. We have the same issue for freezer rack positions, which is also
problematic because putting a barcode label on the rim breaks the seal on
the freezer. We would like to get rid of all these physical barcodes for
positions and have standardized way of creating a variety of positions in
different container types. E.g., a form which allows the user to choose the
number of rows and number of columns needed for a particular application,
just like the "insert table" function in Word.

**Comment by KyndallH** [comment]
We have smaller racks and we ended barcoding them with the small cryovial barcodes.

Do we freezer folks and Dusty want to chat sometime?? I'd be up for a barcoding/freezer chat to see how you are doing it, show how we do it, and see how we can maybe do it better in Arctos.

**Comment by dustymc** [comment]
> do not have positions

Now I'm lost again.

Yes, let's set up a discussion. I suspect ya'll can learn some stuff from each other, and I can definitely write better code when I know a bit more about what actually happens when the scanners come out. @ccicero should get an invite - I'm not sure who all else is doing this.

**Comment by KyndallH** [comment]
Doodle poll on when we can meet.
https://doodle.com/poll/fwym2vqafvmhf3z7
Answer by Thur. so we can set a date/time and we can invite others at AWG meeting to join us if interested.

**Comment by Jegelewicz** [comment]
> This is for racks which DO NOT have position barcodes and hold freezer boxes - right?

Yes.

**Comment by dustymc** [comment]
blocked by https://github.com/ArctosDB/arctos/issues/1743

**Comment by dustymc** [comment]
This is irrelevant with #1743
